CRM/Core/Payment.php

```diff
@@ -1450,7 +1450,7 @@ public function subscriptionURL($entityID = NULL, $entity = NULL, $action = 'can
       FROM civicrm_contribution_recur rec
 INNER JOIN civicrm_contribution con ON ( con.contribution_recur_id = rec.id )
      WHERE rec.id = %1
-  GROUP BY rec.id";
+  GROUP BY rec.id, con.contact_id";
```

Comment:
only issue here would be is if a contribution in the civicrm_contribution table can be linked to multiple contacts

Reply:
@seamuslee001 I would rather say what if we change ```SELECT MAX(con.contact_id)``` ? @eileenmcnaughton will there be any consequence?

Reply:
guess so or should it be min(con.contact_id)

Reply:
MAX would be to fetch the contact of latest contribution linked to a recurring, wouldn't that be correct?

Reply:
which should AFAIK be the same as the first given its a series right?

Reply:
Hmm right ... let's wait for Eileen's reply.

Reply:
Ok so looking at this the point of that join is to filter out recurring transactions where there are no contributions. However, it's not using that as a criteria to determine whether to show the url or anything that seems meaningful to how it's treated. Instead it's being used to determine whether to include a checksum. I think the right answer is to drop the INNER join and the group by

Reply:
so do where IN on the civicrm_contribution table?